### PR TITLE
DCOS-10470: [10/10] Introducing the "Other" section

### PR DESCRIPTION
**Pull request by wavesoft**
---
~~:warning: This PR depends on #1549~~

---

Final parts of the DSL

image

### CODE REVIEW

**Comment by wavesoft** [comment]
retest this please

**Comment by wavesoft** [comment]
let's try to retest this again please

**Comment by wavesoft** [comment]
let's try again to retest this please

**Comment by wavesoft** [comment]
Rebased

**Comment by mlunoe** [comment]
Found this issue:


Steps to reproduce:
- Click filter dropdown
- Check 'Volumes' 
- Check 'Pod'
- Uncheck 'Volumes'

**Comment by mlunoe** [comment]
Code looks good though, and works great otherwise!

**Comment by rcorral** [comment]
@wavesoft I think the "reset" button should work when you're in a bad state. Even if that means clearing everything. Otherwise the user still has to do it themselves. What do you think?
cc/ @leemunroe 
img

**Comment by wavesoft** [comment]
@rcorral ok, I thought that "Reset" was "Undo my changes so far", meaning that it will bring the UI to the state it was before the dropdown. However yes, that should be addressed.

**Comment by orlandohohmeier** [comment]
**mesosphere-ci** retest this please 🙏

**Comment by Poltergeist** [comment]
**mesosphere-ci** retest this please

**Comment by wavesoft** [comment]
let's retest this please again

File: src/js/utils/DSLUpdateUtil.js [hunk 1]
```diff
@@ -3,11 +3,74 @@ import DSLCombinerTypes from '../constants/DSLCombinerTypes';
...
+    src = src.replace(/^\s*,/g, ''); // Orphan commas (left-side)
+    src = src.replace(/,\s*$/g, ''); // Orphan commas (right-side)
+    src = src.replace(/\(\s*,\s*/g, '('); // Orphan commas (left-paren)
+    src = src.replace(/\s*,\s*\)/g, ')'); // Orphan commas (right-paren)
+    src = src.replace(/:,+/g, ':'); // Orphan commas (multi-value)
+    src = src.replace(/\s+,\s+/g, ', '); // Commas with surrounding whitespace
+
+    // Put strings back
+    /* eslint-disable no-control-regex */
+    src = src.replace(/\x01/g, () => {
```
Comment (Poltergeist) [inline src/js/utils/DSLUpdateUtil.js]: Why do we need this? I mean why do we enable rules if we disable them somewhere in the code?

File: plugins/services/src/js/components/dsl/ServiceOtherDSLSection.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import DSLCombinerTypes from '../../../../../../src/js/constants/DSLCombinerTypes';
+import DSLExpression from '../../../../../../src/js/structs/DSLExpression';
+import DSLExpressionPart from '../../../../../../src/js/structs/DSLExpressionPart';
+import DSLFormWithExpressionUpdates from '../../../../../../src/js/components/DSLFormWithExpressionUpdates';
+import DSLUtil from '../../../../../../src/js/utils/DSLUtil';
+import FieldInput from '../../../../../../src/js/components/form/FieldInput';
+import FieldLabel from '../../../../../../src/js/components/form/FieldLabel';
+import FormGroup from '../../../../../../src/js/components/form/FormGroup';
+
+const PARTS = {
```
Comment (Poltergeist) [inline plugins/services/src/js/components/dsl/ServiceOtherDSLSection.js]: This is probably a leftover. Please rename PARTS to something more catchy.

File: src/js/utils/DSLUpdateUtil.js (hunk 1 above)
Comment (wavesoft) [inline src/js/utils/DSLUpdateUtil.js]: > Control characters are special, invisible characters in the ASCII range 0-31. These characters are rarely used in JavaScript strings so a regular expression containing these characters is most likely a mistake.

Not in this particular case. Non-visible character `\x01` is used as a special placeholder for extracted string expressions.

File: plugins/services/src/js/components/dsl/ServiceOtherDSLSection.js (hunk 2 above)
Comment (wavesoft) [inline plugins/services/src/js/components/dsl/ServiceOtherDSLSection.js]: Whoops, indeed. Fixing

**Comment by wavesoft** [comment]
retest this please

**Pull request merged**